Training2/src/task123/task/src/components/progress-bar-rfc/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {useEffect, useState} from 'react';
+
+type ProgressBarProps = {
+    width: number;
+    progress: number
+    onTrigger: (progress: number) => void
+}
+
+const ProgressBar = (props: ProgressBarProps) => {
+    const [barWidth, setBarWidth] = useState<number>(props.width);
+    const [isStop, setIsStop] = useState<boolean>(false);
+
+    useEffect(() => {
+        if(barWidth !== 0 && !isStop) {
+            setTimeout(() => {
```

Comment:
how can we stop this setTimeout when the user clicks on Stop button, we expected that it the counting should be stopped immediately?